src/ModelContextProtocol-SemanticKernel/Extensions/KernelExtensions.Vscode.cs

```diff
@@ -0,0 +1,124 @@
+    public static async Task<KernelPluginCollection> AddToolsFromVsCodeConfigAsync(
+        this KernelPluginCollection plugins,
+        string path,
+        IEnumerable<VSCodeInput> inputs = null,
+        ILoggerFactory? loggerFactory = null,
+        CancellationToken cancellationToken = default)
+    {
+        var config = JsonSerializer.Deserialize<VsCodeSettings>(File.OpenRead(path));
+        return await ProcessVsCodeConfiguration(plugins, config?.McpSection, cancellationToken);
+    }
+
+    /// <summary>
+    /// Adds Model Content Protocol plugins from the Visual Studio settings file (<c>settings.json</c>) and adds it into the plugin collection.
+    /// </summary>
+    /// <param name="plugins">The plugin collection to which the new plugin should be added.</param>
+    /// <param name="instanceType">In case of VSCode Insiders, default can be overriden to auto-detect settings location</param>
+    /// <param name="inputs">Provide non-interactive values for inputs, VS Code would prompt user for it</param>
+    /// <param name="loggerFactory">The optional <see cref="ILoggerFactory"/>.</param>
+    /// <param name="cancellationToken">The optional <see cref="CancellationToken"/>.</param>
+    /// <returns><see cref="KernelPluginCollection"/> containing the functions provided in plugins.</returns>
+    public static async Task<KernelPluginCollection> AddToolsFromVsCodeConfigAsync(
+        this KernelPluginCollection plugins,
+        VsCodeInstanceType instanceType = VsCodeInstanceType.VisualStudioCode,
+        IEnumerable<VSCodeInput> inputs = null,
+        ILoggerFactory? loggerFactory = null,
+        CancellationToken cancellationToken = default)
+    {
+        // https://code.visualstudio.com/docs/configure/settings#_user-settingsjson-location
+        var instancePath = instanceType switch
+        {
+            VsCodeInstanceType.VisualStudioCode => "Code",
+            VsCodeInstanceType.VisualStudioCodeInsiders => "Code - Insiders", // no docs for this, but it's the name on local machine (windows, including white spaces)
+            _ => throw new ArgumentException("Visual Studio Instance Type not supported yet.", nameof(instanceType))
+        };
+
+        var appDataRoaming = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
+        var configPath = Path.Combine(appDataRoaming, instancePath, "User", "settings.json");
+        if (!File.Exists(configPath))
+        {
+            // TODO: Add logging that configuration file was not found
+            return [];
+        }
+
+        var config = JsonSerializer.Deserialize<VsCodeSettings>(File.OpenRead(configPath));
+        return await ProcessVsCodeConfiguration(plugins, config?.McpSection, cancellationToken);
+    }
+
+    internal static async Task<KernelPluginCollection> ProcessVsCodeConfiguration(KernelPluginCollection plugins, VSCodeConfig configuration, CancellationToken cancellationToken = default)
```

Comment:
for private methods, I always make the `CancellationToken cancellationToken` mandatory